drivers/hmis/lib/tasks/one_time_migration_20241111.rake

```diff
@@ -0,0 +1,75 @@
+desc 'One time data migration to populate created_by and updated_by on Custom Assessments'
+# rails driver:hmis:migrate_assessments_20241111
+task migrate_assessments_20241111: [:environment] do
+  assessments = Hmis::Hud::CustomAssessment.joins(:user).preload(:versions)
+  total_records = assessments.count
+  updated_count = 0
+
+  user_ids = Set.new
+  data_source_ids = Set.new
+
+  assessments.find_each.with_index(1) do |assessment, index|
+    # use array methods since this is preloaded
+    versions = assessment.versions.to_a.filter do |version|
+      version.clean_true_user_id || version.clean_user_id
+    end
+
+    unless versions.blank?
+      update_version = versions.max_by(&:created_at)
+      create_version = versions.detect { |v| v.event == 'create' }
+
+      attrs = {}
+
+      attrs[:updated_by_user_id] = (update_version.clean_true_user_id || update_version.clean_user_id) # update_version will always be present
+
+      if create_version
+        attrs[:created_by_user_id] = (create_version.clean_true_user_id || create_version.clean_user_id)
+        user_ids << (create_version.clean_true_user_id || create_version.clean_user_id)
+        data_source_ids << assessment.data_source_id
+      end
+
+      # use `update_columns` to bypass paper trail and timestamp updates
+      assessment.update_columns(**attrs)
+
+      updated_count += 1
+    end
+
+    if index % 1000 == 0 || index == total_records
+      puts "Processed #{index} of #{total_records}. Updated #{updated_count}."
+    end
+  end
+
+  # explicitly do not support multiple data sources
+  raise 'unexpectedly found multiple HMIS data sources' if data_source_ids.size > 1
+
+  # find (and if needed, save) a HUD user for each app user we encountered
+  app_user_ids_to_hud_users = {}
+  user_ids.each do |user_id|
+    user = Hmis::User.find(user_id)
+    user.hmis_data_source_id = data_source_ids.first
+    hud_user = Hmis::Hud::User.from_user(user)
```

Comment:
seems okay to just save these here since there shouldn't too many user records; then we don't need the `Hmis::Hud::User.import(users_to_import)` logic below.

```suggestion
    hud_user = Hmis::Hud::User.from_user(user)
    hud_user.save!
```

Reply:
It turns out calling `save!` is not needed since `Hmis::Hud::User.from_user(user)` calls `first_or_create`, not `first_or_initialize`.